bolt/lib/Passes/PAuthGadgetScanner.cpp

```diff
@@ -438,27 +469,19 @@ void Analysis::runOnFunction(BinaryFunction &BF,
     BF.dump();
   });
 
-  if (BF.hasCFG()) {
-    PacRetAnalysis PRA(BF, AllocatorId, {});
-    FunctionAnalysisResult FAR = computeDfState(PRA, BF, AllocatorId);
-    if (!FAR.RegistersAffected.empty()) {
-      // Redo the analysis, but now also track which instructions last wrote
-      // to any of the registers in RetRegsWithGadgets, so that better
-      // diagnostics can be produced.
-      std::vector<MCPhysReg> RegsToTrack;
-      for (MCPhysReg R : FAR.RegistersAffected)
-        RegsToTrack.push_back(R);
-      PacRetAnalysis PRWIA(BF, AllocatorId, RegsToTrack);
-      FAR = computeDfState(PRWIA, BF, AllocatorId);
-    }
+  if (!BF.hasCFG())
+    return;
 
-    // `runOnFunction` is typically getting called from multiple threads in
-    // parallel. Therefore, use a lock to avoid data races when storing the
-    // result of the analysis in the `AnalysisResults` map.
-    {
-      std::lock_guard<std::mutex> Lock(AnalysisResultsMutex);
-      AnalysisResults[&BF] = FAR;
-    }
+  FunctionAnalysisResult FAR = computeDfState(BF, AllocatorId);
```

Comment:
This is not an issue introduced by this patch, but now seeing this line again, it seems to me that `computeDfState` is no longer a good name for the function called here.
It does more than computing the data flow state on `BF`: it also post-processes the data flow state and return all diagnostics based on the data flow analysis.
Again, not something that has to be fixed in this PR, but if you have ideas for a better name, it would be good to change the name to something better at some point.

Reply:
I will try to came up with a better code structure around this function (and name the resulting functions appropriately) - after all, it is this PR that moves almost everything from `runOnFunction` to `computeDfState` (which is basically "scan the function" now), but deleting `computeDfState` and simply moving everything to `runOnFunction` instead looks like even worse solution.

With the changes made by this PR, it should be possible to replace `computeDfState` with something like "find gadgets" and "augment reports with context information" parts, I think.

Reply:
Reorganized the code a bit - aside from renaming `computeDfState`, this should improve readability in general.

Reply:
Thanks, that looks again much easier to understand, even if it were relatively simple changes.